magenta-lib/src/main/scala/magenta/deployment_type/DeploymentType.scala

```diff
@@ -53,12 +53,18 @@ case class Param[T](name: String,
 
   def get(pkg: DeploymentPackage)(implicit reads: Reads[T]): Option[T] =
     pkg.pkgSpecificData.get(name).flatMap(jsValue => Json.fromJson[T](jsValue).asOpt)
-  def apply(pkg: DeploymentPackage)(implicit reads: Reads[T], manifest: Manifest[T]): T =
-    get(pkg).orElse(defaultValue).orElse(defaultValueFromPackage.map(_(pkg))).getOrElse{
+  def apply(pkg: DeploymentPackage)(implicit reporter: DeployReporter, reads: Reads[T], manifest: Manifest[T]): T = {
```

Comment:
To me the noise of `implicit val reporter = resources.reporter` in lots of places hurts expressiveness more than passing it through as a parameter.